File: Java/AR_126.java
import java.util.Scanner;

public class Solution {
	public static boolean divisores_fact(int p, int n) {
		boolean ret=false;

		if ((p==1 || p<=n) && p>=0 && n>=0) ret=true;
		else ret=false;

		return ret;
	}

	public static boolean resuelveCaso(Scanner scanner) {
		long p, n;
		if (!scanner.hasNextLong()) return false;
		p=scanner.nextLong();
		n=scanner.nextLong();
		if (p<=0 && n<=0) return false;
		
		if (divisores_fact((int)p, (int)n)) System.out.println("YES"); 
		else System.out.println("NO");

		return true;
	}

	public static void main(String[] args) {
		Scanner scanner=new Scanner(System.in);
		while (resuelveCaso(scanner));
		
		scanner.close();
	}
}

File: C/100-200/AR_126.c
#include <stdio.h>

int divisores_fact(int p, int n) {
    int ret=0;

    if ((p==1 || p<=n) && p>=0 && n>=0) ret=1;
    else ret=0;

    return ret;
}

int resuelveCaso() {
    long long int p, n;
    scanf("%lld %lld", &p, &n);

    if (p<=0 && n<=0) return 0;
    
    if (divisores_fact(p, n)) printf("YES\n");
    else printf("NO\n");

    return 1;
}

int main() {
    while (resuelveCaso());
    return 0;
}